README.md

```diff
@@ -17,361 +17,55 @@ connection-less reliable interface. This allows NCCL applications to take
-the
-[GitHub Release Page](https://github.com/aws/aws-ofi-nccl/releases).
-The plugin uses GNU autotools for its build system. You can build it
-as follows:
-
-```
-$ ./configure
-$ make
-$ sudo make install
-```
-
-If you want to install the plugin in a custom path, use the `--prefix`
-configure flag to provide the path. You can also point the build to custom
-dependencies with the following flags:
-
-```
-  --with-libfabric=PATH   Path to non-standard libfabric installation
-  --with-cuda=PATH        Path to non-standard CUDA installation
-  --with-mpi=PATH         Path to non-standard MPI installation
-  --with-hwloc=PATH       Path to non-standard HWLOC installation
-```
-
-By default, the configure script attempts to auto-detect whether it is running
-on an AWS EC2 instance, and if so enables AWS-specific optimizations. These
-optimizations can be enabled regardless of build machine with the following
-config option:
-
-```
-  --enable-platform-aws   Enable AWS-specific configuration and optimizations.
-                          (default: Enabled if on EC2 instance)
-```
-
-To enable trace messages for debugging (disabled by default), use the
-following config option:
-
-```
-   --enable-trace         Enable printing trace messages
-```
-
-To enable UBSAN (Undefined Behaviour Sanitizer), use the following config option:
-
-```
-   --enable-ubsan         Enable undefined behaviour checks with UBSAN
-```
-
-To enable memory access checks with ASAN (disabled by default), use
-the following config option:
-
-```
-   --enable-asan           Enable ASAN memory access checks
-```
-
-In case plugin is configured with `--enable-asan` and the executable
-binary is not compiled and linked with ASAN support, it is required to
-preload the ASAN library, i.e., run the application with `export
-LD_PRELOAD=<path to libasan.so>`.
-
-In case plugin is configured with `--enable-asan` and the plugin is
-run within a CUDA application, environment variable `ASAN_OPTIONS`
-needs to include `protect_shadow_gap=0`. Otherwise, ASAN will crash on
-out-of-memory.
-NCCL currently has some memory leaks and ASAN reports memory leaks by
-default on process exit. To avoid warnings on such memory leaks, e.g.,
-only invalid memory accesses are of interest, add `detect_leaks=0` to
-`ASAN_OPTIONS`.
-
-To enable memory access checks with valgrind (disabled by default),
-use the following config option:
+The best way to build the plugin is to start with the latest [release
+package](https://github.com/aws/aws-ofi-nccl/releases).  The plugin
+developers highly discourage customers from building directly from the
+HEAD of a GitHub branch, as releases go through more extensive testing
+than the pre-commit testing on git branches.
 
-```
-   -with-valgrind[=PATH]  Enable valgrind memory access checks
-```
+Each release will include a README.md file, similar to this one, but
+with detailed build instructions and version requirements.
```

Comment:
This seems like a strange distinction to me. Do other projects do this?

I would (naively) prefer the READMEs to be the same between master and releases, with the more detailed information relegated to separate documentation files (e.g., `INSTALL.md`, like in @rajachan's PR).